app/ai/voice/agents/breeze_buddy/observability/tracing.py

```
@@ -0,0 +1,452 @@
+class PipelineObserver(BaseObserver):
+    """
+    Observer that handles pipeline-level events and manages the parent span lifecycle.
+    """
+
+    def __init__(self, parent_span: trace.Span, **kwargs):
+        super().__init__(**kwargs)
+        self._parent_span = parent_span
+        self._register_event_handler("on_pipeline_ended")
+        self._register_event_handler("on_pipeline_cancelled")
+
+    async def _on_pipeline_ended(self, *args, **kwargs):
```

Ruff (0.14.10)

Unused method argument: `args`

(ARG002)